From 9645af9efe15d767f1247617893879bc79b23523 Mon Sep 17 00:00:00 2001
From: Weiwen Chen <cww@rock-chips.com>
Date: Wed, 9 Sep 2020 21:55:19 +0800
Subject: [PATCH] ARM: dts: rv1109: add 38x38 v10 emmc support

This board is the same as rv1126-38x38-v10-emmc.dts,
except camera sensor.

Signed-off-by: Weiwen Chen <cww@rock-chips.com>
Change-Id: Ib0850951b5c4e6cb7d33a71aa8fb4956b03df20a
---
 arch/arm/boot/dts/Makefile                  |  1 +
 arch/arm/boot/dts/rv1109-38x38-v10-emmc.dts | 81 +++++++++++++++++++++
 2 files changed, 82 insertions(+)
 create mode 100644 arch/arm/boot/dts/rv1109-38x38-v10-emmc.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index d2178918cdf6..426be970ade5 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -842,6 +842,7 @@ dtb-$(CONFIG_ARCH_RENESAS) += \
 	sh73a0-kzm9g.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += \
 	rv1108-evb.dtb \
+	rv1109-38x38-v10-emmc.dtb \
 	rv1109-38-v10-spi-nand.dtb \
 	rv1109-ai-cam-ddr3-v1.dtb \
 	rv1109-evb-ddr3-v12-facial-gate.dtb \
diff --git a/arch/arm/boot/dts/rv1109-38x38-v10-emmc.dts b/arch/arm/boot/dts/rv1109-38x38-v10-emmc.dts
new file mode 100644
index 000000000000..da9c9412b51c
--- /dev/null
+++ b/arch/arm/boot/dts/rv1109-38x38-v10-emmc.dts
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Rockchip Electronics Co., Ltd.
+ */
+
+/dts-v1/;
+
+#include "rv1126-38x38-v10-emmc.dts"
+
+/ {
+	model = "Rockchip RV1109 38x38 V10 EMMC DDR3 Board";
+	compatible = "rockchip,rv1109-38x38-v10-emmc", "rockchip,rv1109";
+
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xff570000 console=ttyFIQ0 root=PARTUUID=614e0000-0000 rootfstype=squashfs rootwait snd_aloop.index=7";
+	};
+
+};
+
+&csi_dphy0 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi_in_ucam2: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&ucam_out2>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			csidphy0_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&mipi_csi2_input>;
+			};
+		};
+	};
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <400000>;
+
+	/delete-node/ imx415@1a ;
+
+	imx335: imx335@1a {
+		compatible = "sony,imx335";
+		reg = <0x1a>;
+		clocks = <&cru CLK_MIPICSI_OUT>;
+		clock-names = "xvclk";
+		power-domains = <&power RV1126_PD_VI>;
+		pinctrl-names = "rockchip,camera_default";
+		pinctrl-0 = <&mipicsi_clk0>;
+		avdd-supply = <&vcc3v3_sys>;
+		dovdd-supply = <&vcc_1v8>;
+		dvdd-supply = <&vcc_dvdd>;
+		reset-gpios = <&gpio1 RK_PD5 GPIO_ACTIVE_LOW>;
+		rockchip,camera-module-index = <1>;
+		rockchip,camera-module-facing = "front";
+		rockchip,camera-module-name = "MTV4-IR-E-P";
+		rockchip,camera-module-lens-name = "40IRC-4MP-F16";
+		ir-cut = <&cam_ircut0>;
+		flash-leds = <&flash_ir>;
+		port {
+			ucam_out2: endpoint {
+				remote-endpoint = <&mipi_in_ucam2>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+	};
+};
-- 
2.35.3

